### DISCUSSION TITLE
UInput event is different to input

### DISCUSSION THREAD

**Discussion by CyberSekure** [nuxt/nuxt#26587] [Questions] [answered]
Apologies if I'm totally getting this wrong. Quite new to the world of Nuxt, Vue and Javascript!

I'm trying to create a file handler function that uses the event from a UInput component in a similar way to how I might do with input. e.g. 
```
<UInput
type="file"
accept="image/*"
@change="handlePrimaryImageUpload"
/>
```

vs

```
<input
type="file"
accept="image/*"
@change="handlePrimaryImageUpload"
>
```

However the event from the UInput is a string, whereas the event from the input has all the target and files properties so I can use something like `const file = event.target.files[0];` to access the file.

Am I using UInput incorrectly? I'd prefer to use it than input as it's just a lot easier with keeping the styling consistent.

Any help much appreciated!

**Comment by TheAlexLichter** [comment, marked as answer]
There is a difference between native events and [custom component events](https://vuejs.org/guide/components/events.html#component-events). Native events have an `event` object containing info around where the event originated and the "data", while component events only contain the provided payload and nothing around it.

**Reply by CyberSekure** [reply]
Nice one, much appreciated! I’ll have a read thank you
